codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaders.java

```diff
@@ -1576,32 +1595,64 @@ public boolean contains(String name, String value, boolean ignoreCase) {
      * otherwise a case sensitive compare is run to compare values.
      */
     public boolean containsValue(CharSequence name, CharSequence value, boolean ignoreCase) {
-        List<String> values = getAll(name);
-        if (values.isEmpty()) {
-            return false;
-        }
-
-        for (String v: values) {
-            if (contains(v, value, ignoreCase)) {
-                return true;
+        Iterator<? extends CharSequence> itr = valueCharSequenceIterator(name);
+        if (ignoreCase) {
+            while (itr.hasNext()) {
+                if (containsCommaSeparatedTrimmed(itr.next(), value, true)) {
```

Comment:
why not use the same loop and pass `ignoreCase` into `containsCommaSeparatedTrimmed(....)` ?

Reply:
+1 good catch ... the code structure was previously different where the loop contents were unique but now I can simplify.